File: win-prog/ProgWinCS/Toolbars and Status Bars/SimpleToolBar/SimpleToolBar.cs
//--------------------------------------------
// SimpleToolBar.cs © 2001 by Charles Petzold
//--------------------------------------------
using System;
using System.Drawing;
using System.Windows.Forms;

class SimpleToolBar: Form
{
     public static void Main()
     {
          Application.Run(new SimpleToolBar());
     }
     public SimpleToolBar()
     {
          Text = "Simple Toolbar";

               // Create a simple menu (just for show).

          Menu = new MainMenu();
          Menu.MenuItems.Add("File");
          Menu.MenuItems.Add("Edit");
          Menu.MenuItems.Add("View");
          Menu.MenuItems.Add("Help");

               // Create ImageList.

          Bitmap bm = new Bitmap(GetType(), 
                                 "SimpleToolBar.StandardButtons.bmp");

          ImageList imglst = new ImageList();
          imglst.Images.AddStrip(bm);
          imglst.TransparentColor = Color.Cyan;

               // Create ToolBar.

          ToolBar tbar = new ToolBar();
          tbar.Parent = this;
          tbar.ImageList = imglst;
          tbar.ShowToolTips = true;

               // Create ToolBarButtons.

          string[] astr = {"New", "Open", "Save", "Print", 
                           "Cut", "Copy", "Paste" };

          for (int i = 0; i < 7; i++)
          {
               ToolBarButton tbarbtn = new ToolBarButton();
               tbarbtn.ImageIndex = i;
               tbarbtn.ToolTipText = astr[i];

               tbar.Buttons.Add(tbarbtn);
          }
     }
}

File: win-prog/ProgWinVB/Toolbars and Status Bars/SimpleToolBar/SimpleToolBar.vb
'----------------------------------------------
' SimpleToolBar.vb (c) 2002 by Charles Petzold
'----------------------------------------------
Imports System
Imports System.Drawing
Imports System.Windows.Forms

Class SimpleToolBar
    Inherits Form

    Shared Sub Main()
        Application.Run(New SimpleToolBar())
    End Sub

    Sub New()
        Text = "Simple Toolbar"

        ' Create a simple menu (just for show).
        Menu = New MainMenu()
        Menu.MenuItems.Add("File")
        Menu.MenuItems.Add("Edit")
        Menu.MenuItems.Add("View")
        Menu.MenuItems.Add("Help")

        ' Create ImageList.
        Dim bm As New Bitmap(Me.GetType(), "StandardButtons.bmp")

        Dim imglst As New ImageList()
        imglst.Images.AddStrip(bm)
        imglst.TransparentColor = Color.Cyan

        ' Create ToolBar.
        Dim tbar As New ToolBar()
        tbar.Parent = Me
        tbar.ImageList = imglst
        tbar.ShowToolTips = True

        ' Create ToolBarButtons.
        Dim astr() As String = {"New", "Open", "Save", "Print", _
                                "Cut", "Copy", "Paste"}
        Dim i As Integer

        For i = 0 To astr.GetUpperBound(0)
            Dim tbb As New ToolBarButton()
            tbb.ImageIndex = i
            tbb.ToolTipText = astr(i)
            tbar.Buttons.Add(tbb)
        Next i
    End Sub
End Class
